### PR TITLE
Fixed issue #149: bson_init_static(): precondition failed: data

### PR DESCRIPTION
**Pull request by bjori**
https://jira.mongodb.org/browse/PHPC-502

```
<?php                                                                                
$opts = array (                                                                      
  'allowPartialResults' => false,                                                    
  'batchSize' => 101,                                                                
  'comment' => '',                                                                   
  'cursorType' => 0,                                                                 
  'limit' => 1,                                                                      
  'maxTimeMS' => 0,                                                                  
  'modifiers' =>                                                                     
  array (                                                                            
  ),                                                                                 
  'noCursorTimeout' => false,                                                        
  'oplogReplay' => false,                                                            
  'projection' =>                                                                    
  array (                                                                            
  ),                                                                                 
  'skip' => 0,                                                                       
  'sort' =>                                                                          
  array (                                                                            
  ),                                                                                 
  'cursorFlags' => 0,                                                                
);                                                                                   
                                                                                     
$m = new MongoDB\Driver\Manager("mongodb://localhost:27017");                        
$q = new MongoDB\Driver\Query(array(), $opts);                                       
                                                                                     
$m->executeQuery("dbname.collname", $q);          
echo "I'm alive\n";
```

### CODE REVIEW

File: php_phongo.c [hunk 1]
```diff
@@ -329,9 +328,13 @@ bool phongo_query_init(php_phongo_query_t *query, bson_t *filter, bson_t *option
...
 
 			bson_iter_document(&iter, &len, &data);
-			bson_init_static(&tmp, data, len);
-			bson_copy_to_excluding_noinit(&tmp, query->query, "nadastrada", NULL);
-			bson_destroy (&tmp);
+			if (len) {
+				bson_t tmp;
+
+				bson_init_static(&tmp, data, len);
+				bson_copy_to_excluding_noinit(&tmp, query->query, "nadastrada", NULL);
```
Comment (derickr) [inline php_phongo.c]: What does nadastrada mean?

Comment (bjori) [inline php_phongo.c]: "nada" means "nothing", "strada" means "road" and rimes with "nada".

Comment (derickr) [inline php_phongo.c]: I realise that, but what meaning does it have in the code?

Comment (bjori) [inline php_phongo.c]: nothing. It can be any value, its not used.

Comment (derickr) [inline php_phongo.c]: OK - that needs to be changed to something descriptive then.

**Comment by derickr** [comment]
Merged. Thanks!

**Pull request closed without merging**